bench-exchange/src/bench.rs

```diff
@@ -1067,24 +1085,31 @@ mod tests {
                 error!("Failed to discover {} nodes: {:?}", NUM_NODES, err);
                 exit(1);
             });
-        if nodes.len() < NUM_NODES {
-            error!(
-                "Error: Insufficient nodes discovered.  Expecting {} or more",
-                NUM_NODES
-            );
-            exit(1);
-        }
         let client_ctors: Vec<_> = nodes
             .iter()
-            .map(|node| {
+            .filter_map(|node| {
                 let cluster_entrypoint = node.clone();
                 let cluster_addrs = cluster_entrypoint.client_facing_addr();
-                let client_ctor =
-                    move || -> ThinClient { create_client(cluster_addrs, FULLNODE_PORT_RANGE) };
-                client_ctor
+                if ContactInfo::is_valid_address(&cluster_addrs.0)
```

Comment:
Sloppy to duplicate all this code in two modules in the same crate in the same commit.  👎